tests/dhcp_relay/test_dhcp_relay.py

```diff
@@ -55,6 +56,78 @@ def check_interface_status(duthost):
+    return total_counters
+
+
+def validate_dhcpcom_counters(dhcp_relay, duthost):
+    """Validate the dhcpcom counters"""
+    downlink_vlan_iface = dhcp_relay['downlink_vlan_iface']['name']
+    uplink_portchannels_interfaces = dhcp_relay['uplink_interfaces']
+    client_iface = dhcp_relay['client_iface']['name']
+    dhcp_server_sum = len(dhcp_relay['downlink_vlan_iface']['dhcp_server_addrs'])
+    portchannels = dhcp_relay['portchannels']
+    uplink_interfaces = []
+    for portchannel_name in uplink_portchannels_interfaces:
+        if portchannel_name in portchannels.keys():
+            uplink_interfaces.extend(portchannels[portchannel_name]['members'])
+        else:
+            uplink_interfaces.append(portchannel_name)
+
+    vlan_interface_counter = query_vlan_interface_counters(duthost, downlink_vlan_iface, [])
+    client_interface_counter = query_vlan_interface_counters(duthost, downlink_vlan_iface, [client_iface])
+    uplink_portchannels_interfaces_counter = query_vlan_interface_counters(
+        duthost, downlink_vlan_iface, uplink_portchannels_interfaces
+    )
+    uplink_interface_counter = query_vlan_interface_counters(duthost, downlink_vlan_iface, uplink_interfaces)
+
+    assert vlan_interface_counter == client_interface_counter
+    assert uplink_interface_counter == uplink_portchannels_interfaces_counter
+    vlan_interface_counter_expected_value = (
+        '{"RX": {"Ack": 0, "Bootp": 1, "Decline": 0, "Discover": 1, "Inform": 0, "Nak": 0, "Offer": 0, '
+        '"Release": 0, "Request": 2, "Unknown": 0}, "TX": {"Ack": 1, "Bootp": 0, "Decline": 0, '
+        '"Discover": 0, "Inform": 0, "Nak": 0, "Offer": 1, "Release": 0, "Request": 0, "Unknown": 0}}'
+    )
+    uplink_interface_counter_expected_value = (
+        '{"RX": {"Ack": 1, "Bootp": 0, "Decline": 0, "Discover": 0, "Inform": 0, "Nak": 0, "Offer": 1, '
+        '"Release": 0, "Request": 0, "Unknown": 0}, "TX": {"Ack": 0, "Bootp": %d, "Decline": 0, "Discover": %d, '
+        '"Inform": 0, "Nak": 0, "Offer": 0, "Release": 0, "Request": %d, "Unknown": 0}}'
+    ) % (dhcp_server_sum, dhcp_server_sum, dhcp_server_sum * 2)
+
+    assert json.dumps(vlan_interface_counter, sort_keys=True) == vlan_interface_counter_expected_value, \
+        "VLAN interface counters do not match expected value. Expected: {}, Actual: {}".format(
+            json.dumps(vlan_interface_counter, sort_keys=True), vlan_interface_counter_expected_value
+        )
+    assert json.dumps(uplink_interface_counter, sort_keys=True) == uplink_interface_counter_expected_value, \
+        "Uplink interface counters do not match expected value. Expected: {}, Actual: {}".format(
+            json.dumps(uplink_interface_counter, sort_keys=True), uplink_interface_counter_expected_value
+        )
+
+
+def init_and_verify_dhcp_relay_counters(duthost):
+    command_output = duthost.shell("sudo sonic-clear dhcp_relay ipv4 counters")
+    assert command_output["failed"] is False
```

Comment:
Actually, no need this assert here. Because you didn't use ignore_module_errors=True in `shell` function, then if the cmd run failed, it would directly raise an exception